### DISCUSSION TITLE
How to handle middleware if not using src/app folder?

### DISCUSSION THREAD

**Discussion by cerori** [vercel/next.js#50026] [Help] [answered]
### Summary

If I write middleware.ts under the src folder, it works fine. But if I write middleware.ts under app folder without src folder, it doesn't work. What is the solution?

**Comment by devjiwonchoi** [comment, marked as answer]
The middleware.ts should be placed in the root of your project, which should be at the same level as the `app` folder.
In your case, the middleware.ts should be inside the `src` folder which will be the same level as your `app` folder.

> [Convention](https://nextjs.org/docs/app/building-your-application/routing/middleware#convention)
Use the file middleware.ts (or .js) in the root of your project to define Middleware. For example, at the same level as pages or app, or inside src if applicable.

## Reference
- App Router Middleware: https://nextjs.org/docs/app/building-your-application/routing/middleware

**Reply by cerori** [reply]
thank you
